lib/charms/layer/snap.py

```diff
@@ -105,6 +105,85 @@ def connect_all():
             connect(plug, slot)
 
 
+def disable(snapname):
+    '''Disables a snap in the system
+
+    Sets the snap.disabled.{snapname} flag
+
+    This method doesn't affect any snap state if requested snap does not
+    exist
+    '''
+    hookenv.log('Disabling {} snap'.format(snapname))
+    if not reactive.is_flag_set('snap.installed.{}'.format(snapname)):
+        hookenv.log(
+            'Cannot disable {} snap because it is not installed'.format(
+                snapname), hookenv.WARNING)
```

Comment:
Needs a 'return' after the log message. Here and the others.

Reply:
ouch!. Done